grpc-zpages/docker/envoy/zprox.sh

```diff
@@ -56,6 +58,27 @@ static_resources:
                 route:
                   cluster: local_service_web
           http_filters:
+          - name: envoy.lua
+            config:
+              inline_code: |
+                function envoy_on_request(request_handle)
+                  -- hyphens are special characters in lua search patterns and must be escaped with a %
+                  function esc_hyphens(s)
+                    result, _ = string.gsub(s, "%-", "%%%-")
+                    return result
+                  end
+                  headers = request_handle:headers()
+                  content_type = headers:get("content-type")
+                  if content_type and string.find(content_type, esc_hyphens("application/grpc-web")) then
+                    xsrf_header = headers:get("$GDEBUG_TOKEN_NAME")
+                    xsrf_cookie = string.match(headers:get("cookie") or "", esc_hyphens("$GDEBUG_TOKEN_NAME=(%w+)"))
+                    -- ~= means "not equals" in lua
+                    if xsrf_header == nil or xsrf_header == "" or xsrf_cookie == nil or xsrf_cookie == "" or xsrf_header ~= xsrf_cookie then
```

Comment:
This might be asking a lot, and feel free to decline, but I have always used the constant time comparison here.  Not sure if it matters.

Reply:
It may not be so important here since no secret values are being compared. FWIW strings in lua are interned so the comparison itself is constant time.